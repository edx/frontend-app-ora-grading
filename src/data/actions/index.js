import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';
import submissions from './submissions';

export default StrictDict({
  app,
  grading,
  submissions,
});
