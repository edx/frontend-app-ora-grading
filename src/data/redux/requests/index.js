export { actions, reducer } from './reducer';
