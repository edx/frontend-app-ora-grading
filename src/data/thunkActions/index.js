import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';

export default StrictDict({
  app,
  grading,
});
