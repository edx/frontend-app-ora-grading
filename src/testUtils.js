// eslint-disable-next-line import/prefer-default-export
export const formatMessage = (msg, values) => {
  let message = msg.defaultMessage;
  if (values === undefined) { return message; }
  Object.keys(values).forEach(key => {
    // eslint-disable-next-line
    message = message.replace(`{${key}}`, values[key]);
  });
  return message;
};
