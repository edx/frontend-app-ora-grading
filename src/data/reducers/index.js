import { combineReducers } from 'redux';

import app from './app';
import submissions from './submissions';

/* istanbul ignore next */
const rootReducer = combineReducers({
  app,
  submissions,
});

export default rootReducer;
