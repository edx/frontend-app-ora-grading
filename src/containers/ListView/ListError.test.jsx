import React from 'react';
import { shallow } from 'enzyme';

import { selectors } from 'data/redux';

import { formatMessage } from 'testUtils';
import {
  ListError,
  mapStateToProps,
} from './ListError';

jest.mock('data/redux', () => ({
  selectors: {
    app: {
      courseId: (...args) => ({ courseId: args }),
    },
  },
  thunkActions: {
    grading: {
      loadSelectionForReview: (...args) => ({ loadSelectionForReview: args }),
    },
  },
}));

jest.mock('data/services/lms/urls', () => ({
  openResponse: (courseId) => `api/openResponse/${courseId}`,
}));

let el;
jest.useFakeTimers('modern');

describe('ListError component', () => {
  describe('component', () => {
    const props = {
      courseId: 'test-course-id',
    };
    beforeEach(() => {
      props.loadSelectionForReview = jest.fn();
      props.intl = { formatMessage };
    });
    describe('render tests', () => {
      beforeEach(() => {
        el = shallow(<ListError {...props} />);
      });
      test('snapshot', () => {
        expect(el).toMatchSnapshot();
      });
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { some: 'test-state' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('courseId loads from app.courseId', () => {
      expect(mapped.courseId).toEqual(selectors.app.courseId(testState));
    });
  });
});
