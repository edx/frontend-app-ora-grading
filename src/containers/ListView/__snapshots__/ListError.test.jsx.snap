// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListError component component render tests snapshot 1`] = `undefined`;
